samples/excel/45-named-item/create-and-use-named-item-for-range.yaml

```diff
@@ -0,0 +1,171 @@
+id: excel-create-and-use-named-item-for-range
+name: Create and use named range item
+description: Create and use named range item
+author: OfficeDev
+host: EXCEL
+api_set:
+    ExcelApi: 1.4
+script:
+    content: |-
+        $("#add-name").click(() => tryCatch(addName));
+        $("#format-named-range").click(() => tryCatch(formatRangeByName));
+        $("#setup").click(() => tryCatch(setup));
+
+        async function addName() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                const headerRange = sheet.getRange("A1:E1");
+                sheet.names.add("ExpensesHeader", headerRange);
+                const namedItems = sheet.names.load("name, type");
+
+                await context.sync();
+
+                let namedItemsList = `This workbook contains ${namedItems.items.length} named item(s):`;
+                for (let i = 0; i < namedItems.items.length; i++) {
+                    namedItemsList += JSON.stringify(namedItems.items[i]);
+                }
+                OfficeHelpers.UI.notify(namedItemsList);
+
+                await context.sync();
+            });
+        }
+
+        async function formatRangeByName() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                const headerRowName = sheet.names.getItem("ExpensesHeader");
+                const headerRange = headerRowName.getRange();
+                headerRange.format.fill.color = "red";
+
+                await context.sync();
+            });
+        }
+
+        async function setup() {
+            await Excel.run(async (context) => {
+                const sheet = await OfficeHelpers.ExcelUtilities.forceCreateSheet(context.workbook, "Sample");
+                const expensesTable = sheet.tables.add("A1:D1", true);
+                expensesTable.name = "ExpensesTable";
+                expensesTable.getHeaderRowRange().values = [["Date", "Merchant", "Category", "Amount"]];
+                const newData = transactions.map(item =>
+                    [item.date, item.merchant, item.category, item.amount]);
+                expensesTable.rows.add(null, // Row position (null = at the bottom)
+                                       newData);
+                sheet.getUsedRange().format.autofitColumns();
```

Comment:
You shouldn't be re-getching the same object (`getUsedRange()` twice, it's wasteful).  Always conserve your `get` methods.  Set to a usedRange variable instead.